data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderServerProxy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder.server;
+
+import com.linecorp.armeria.server.Server;
+import org.opensearch.dataprepper.peerforwarder.PeerForwarderConfiguration;
+
+public class PeerForwarderServerProxy implements PeerForwarderServer {
+    private final PeerForwarderServer peerForwarderServer;
+
+    public PeerForwarderServerProxy(final PeerForwarderConfiguration peerForwarderConfiguration, final Server server) {
+
+        // update this conditional based on PeerForwarderProvider
+        if (true) {
```

Comment:
I believe that we will need to determine this value when `start()` is called.

Reply:
I think we've to move this out of constructor as `getRegisteredPeerForwarders()` will return zero during bean creation.